src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs

```diff
@@ -31,8 +30,16 @@ public static AuthenticationBuilder AddScriptJwtBearer(this AuthenticationBuilde
                             {
                                 OnMessageReceived = c =>
                                 {
+                                    // By default, tokens are passed via the standard Authorization Bearer header. However we also support
+                                    // passing tokens via the x-ms-site-token header.
+                                    if (c.Request.Headers.TryGetValue(ScriptConstants.SiteTokenHeaderName, out StringValues values))
+                                    {
+                                        // the token we set here will be the one used - Authorization header won't be checked.
+                                        c.Token = values.FirstOrDefault();
```

Comment:
Does this have any impact of lines 43 to 46? The `CreateTokenValidationParameters();`. I don't know the precise behavior of auth middleware, but I am guessing by us setting `c.Token` here, we will short-circuit the default behavior of extracting a token from `Authorization` header. I want to make sure what looks like validation customization below still operates as expected in this scenario.

Reply:
The authenticate method for this middleware checks whether a token value has been specified [here](https://github.com/aspnet/Security/blob/26d27d871b7992022c082dc207e3d126e1d9d278/src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs#L63) and uses it if so. We're using this extensibility as designed.